frontend/src/views/Transactions/Transactions.jsx

```diff
@@ -0,0 +1,185 @@
+import BCAlert from '@/components/BCAlert'
+import BCButton from '@/components/BCButton'
+import { DownloadButton } from '@/components/DownloadButton'
+import { ROUTES } from '@/constants/routes'
+import { useApiService } from '@/services/useApiService'
+import { faCirclePlus } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Box, Typography } from '@mui/material'
+import { AgGridReact } from 'ag-grid-react'
+import * as dayjs from 'dayjs'
+import { useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { dummy } from './data'
+import { getStatus } from '@/utils/getStatus'
+import { getOrgnization } from '@/utils/getOrganization'
+
+export const Transactions = () => {
+  const navigate = useNavigate()
+  const apiService = useApiService()
+
+  const [isDownloadingTransactions, setIsDownloadingTransactions] =
+    useState(false)
+  const [alertMessage, setAlertMessage] = useState('')
+  const [alertSeverity, setAlertSeverity] = useState('info')
+
+  const handleDownloadTransactions = async () => {
+    alert('Download Transactions clicked')
+
+    // enable once we have transactions in db
+
+    // setIsDownloadingTransactions(true)
+    // setAlertMessage('')
+    // try {
+    //   await apiService.download('/transactions/export')
+    //   isDownloadingTransactions(false)
+    // } catch (error) {
+    //   console.error('Error downloading organization information:', error)
+    //   isDownloadingTransactions(false)
+    //   setAlertMessage('Failed to download organization information.')
+    //   setAlertSeverity('error')
+    // }
+  }
+
+  return (
+    <>
+      <div>
+        {alertMessage && (
+          <BCAlert data-test="alert-box" severity={alertSeverity}>
+            {alertMessage}
+          </BCAlert>
+        )}
+      </div>
+      <Typography variant="h5" mb={2}>
+        Transactions
+      </Typography>
+      <Box display={'flex'} gap={2} mb={2}>
+        <BCButton
+          variant="contained"
+          size="small"
+          color="primary"
+          startIcon={<FontAwesomeIcon icon={faCirclePlus} />}
+          onClick={() => navigate(ROUTES.TRANSACTIONS_ADD)}
+        >
+          <Typography variant="subtitle2">New Transfer</Typography>
+        </BCButton>
+        <DownloadButton
+          onDownload={handleDownloadTransactions}
+          isDownloading={isDownloadingTransactions}
+          label="Download as .xls"
+          downloadLabel="Downloading Transaction Information..."
+          dataTest="download-transactions-button"
+        />
+      </Box>
+      <AgGridReact
+        className="ag-theme-alpine"
+        animateRows="true"
+        columnDefs={[
```

Comment:
can you please move the column Definition to a schema.js file for simplicity? Also if you need assistance with BCDataGridServer please let me know I can write documentation if it further helps.

Reply:
I've moved the coldefs option out of the component. The coldefs are only used in this component so leaving it in here would make sense. what do you think?

Reply:
if you start adding any valueGetters or Renders it could grow in size, so separating the functionality would make it easy if we want to make changes to table details or endpoints. We could have more discussion with Alex during our dev session if you strongly feel that it should be in the same file.

Reply:
thats true, to future proof ill move options to its own file